src/sentry/static/sentry/app/components/charts/pieChart.jsx

```diff
@@ -1,12 +1,71 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 
+import Legend from './components/legend';
 import PieSeries from './series/pieSeries';
 import BaseChart from './baseChart';
 
 class PieChart extends React.Component {
   static propTypes = {
     // We passthrough all props exception `options`
     ...BaseChart.propTypes,
+
+    // Attempt to select first series in chart (to show in center of PieChart)
+    selectOnRender: PropTypes.bool,
+  };
+
+  constructor(props) {
+    super(props);
+    this.chart = React.createRef();
+    this.isInitialSelected = true;
+    this.selected = 0;
+  }
+
+  componentDidMount() {
+    let {selectOnRender} = this.props;
```

Comment:
Do we always destructure to leverage PropTypes checking?

Reply:
It's unrelated to PropTypes checking (PropTypes are checked only in dev and get stripped in prod, and done at render time I believe).

It's mostly a style thing and I think it helps readability since you don't see `this.props` used everywhere.